tools/lib/optimizeB3dm.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+var Cesium = require('cesium');
+var GltfPipeline = require('gltf-pipeline');
+var Promise = require('bluebird');
+var zlib = require('zlib');
+var isGzipped = require('./isGzipped');
+
+var zlibGunzip = Promise.promisify(zlib.gunzip);
+
+var Cartesian3 = Cesium.Cartesian3;
+var DeveloperError = Cesium.DeveloperError;
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+
+var Pipeline = GltfPipeline.Pipeline;
+var addCesiumRTC = GltfPipeline.addCesiumRTC;
+var getBinaryGltf = GltfPipeline.getBinaryGltf;
+var parseBinaryGltf = GltfPipeline.parseBinaryGltf;
+
+module.exports = optimizeB3dm;
+
+/**
+ * Given an input, possibly gzipped buffer containing a b3dm, optimize it using gltf-pipeline with the provided options
+ *
+ * @param {Buffer} buffer The buffer containing the b3dm.
+ * @param {Object} [options] Options specifying custom gltf-pipeline behavior.
+ * @returns {Buffer} The optimized b3dm buffer.
+ */
+function optimizeB3dm(buffer, options) {
```

Comment:
Can we separate this into explicit functions:
* b3dm to glTF (and provide this as a command-line tool)
* Running gltf-pipeline
* glTF to b3dm (and provide this as a command-line tool)
   * There should be an option to pass in the parameters needed for the header and batch table